solutions/Fully configurable/main.tf

```diff
@@ -0,0 +1,235 @@
+
+locals {
+  parsed_existing_secrets_manager_crn = var.existing_secrets_manager_crn != null ? split(":", var.existing_secrets_manager_crn) : []
+  secrets_manager_guid                = var.existing_secrets_manager_crn != null ? (length(local.parsed_existing_secrets_manager_crn) > 0 ? local.parsed_existing_secrets_manager_crn[7] : null) : module.secrets_manager.secrets_manager_guid
+  secrets_manager_crn                 = var.existing_secrets_manager_crn != null ? var.existing_secrets_manager_crn : module.secrets_manager.secrets_manager_crn
+  secrets_manager_region              = var.existing_secrets_manager_crn != null ? (length(local.parsed_existing_secrets_manager_crn) > 0 ? local.parsed_existing_secrets_manager_crn[5] : null) : module.secrets_manager.secrets_manager_region
+  sm_endpoint_type                    = var.existing_secrets_manager_crn != null ? var.existing_secrets_endpoint_type : var.allowed_network == "private-only" ? "private" : "public"
+}
+
+module "secrets_manager" {
+  depends_on               = [time_sleep.wait_for_authorization_policy]
+  source                   = "../.."
+  existing_sm_instance_crn = var.existing_secrets_manager_crn
+  resource_group_id        = var.existing_secrets_manager_crn == null ? module.resource_group[0].resource_group_id : data.ibm_resource_instance.existing_sm[0].resource_group_id
+  region                   = var.region
+  secrets_manager_name     = var.prefix != null ? "${var.prefix}-${var.secrets_manager_instance_name}" : var.secrets_manager_instance_name
+  sm_service_plan          = var.service_plan
+  allowed_network          = var.allowed_network
+  sm_tags                  = var.secret_manager_tags
+  # kms dependency
+  kms_encryption_enabled            = var.kms_encryption_enabled
+  existing_kms_instance_guid        = local.existing_kms_guid
+  kms_key_crn                       = local.kms_key_crn
+  skip_kms_iam_authorization_policy = var.skip_kms_iam_authorization_policy || local.create_cross_account_auth_policy
+  # event notifications dependency
+  enable_event_notification        = var.enable_event_notification
+  existing_en_instance_crn         = var.existing_event_notification_instance_crn
+  skip_en_iam_authorization_policy = var.skip_event_notification_iam_authorization_policy
+  endpoint_type                    = local.sm_endpoint_type
+  cbr_rules                        = var.cbr_rules
+}
+
+# Configure an IBM Secrets Manager IAM credentials engine for an existing IBM Secrets Manager instance.
+module "iam_secrets_engine" {
+  count                = var.iam_engine_enabled ? 1 : 0
+  source               = "terraform-ibm-modules/secrets-manager-iam-engine/ibm"
+  version              = "1.2.6"
+  region               = local.secrets_manager_region
+  iam_engine_name      = var.prefix != null ? "${var.prefix}-${var.iam_engine_name}" : var.iam_engine_name
+  secrets_manager_guid = local.secrets_manager_guid
+  endpoint_type        = local.sm_endpoint_type
+}
```

Comment:
Do not call the IAM engine - it is getting deprecated. Instead we need to expose a new s2s auth policy which takes care of this now. It is coming in https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/pull/237 (I will merge once pipeline passes)